File: Rentify.Model/ResponseObjects/LoginResponse.cs
// Rentify.Model/Responses/LoginResponse.cs
using Rentify.Model.ResponseObjects;
using System;
using System.Collections.Generic;

namespace Rentify.Model.ResponseObject
{
    public class LoginResponse
    {
        public int UserId { get; set; }
        public string UserName { get; set; }
        public string Token { get; set; } = string.Empty;
        public bool IsLoggingFirstTime {get; set;}
        public List<String>? Roles { get; set; }
    }
}

File: UI/rentify_desktop/lib/models/login_response.dart
class LoginResponse {
  int userId;
  String userName;
  String token;
  List<String> roles;

  LoginResponse({
    required this.userId,
    required this.userName,
    required this.token,
    required this.roles,
  });

  factory LoginResponse.fromJson(Map<String, dynamic> json) {
    return LoginResponse(
      userId: json["userId"],
      userName: json["userName"],
      token: json["token"],
      roles: List<String>.from(json["roles"]),
    );
  }
}
